### ISSUE TITLE
Strange behavior with CanvasRadialGradientBrush

### ISSUE THREAD

**Issue by AghaKhan** [Microsoft/Win2D#551]
Even I am not working with SVG file but you may conform the expected result with SVG files.
So I created a button.svg file which you find Assets folder. I was expecting that result with Win2D. I created a small project to show the problem. It has few lines of code which is attached.  The result is totally different what I was expecting. 
[Button.zip](https://github.com/Microsoft/Win2D/files/1270794/Button.zip)
<img width="244" alt="outputs" src="https://user-images.githubusercontent.com/5318363/29977634-e9b87186-8ef2-11e7-9d3b-f0bfd44c2c5a.png">

**Comment by shawnhar** [comment]
I don't see a problem with the Win2D rendering here, although it's hard to follow what coordinate systems you are using due to the combination of transform matrix alongside an explicitly specified center point.

I recommend simplifying your gradient configuration to use an identity transform, then just set its center point directly to wherever you want the center of the gradient to be.

You can't expect identical rendering in SVG vs. D2D/Win2D just by copying parameters from one to the other, because these graphics APIs are not entirely the same.  It's possible to do all the same things with gradients in both, but the coordinate systems and ways parameters are specified are sometimes different.

**Comment by AghaKhan** [comment]
Thank you for kind reply.
If you look the picture I provided, you will find, the output of Win2D is not close to SVG file. The direction of the gradient  is different direction.  I removed 5 gradients out of 9 but the result is the same. I am playing with graphics and if the direction is wrong then there is no hope. I also provided the simple sample to isolate the problem. This small sample has one brush and one path only nothing else. The is my humble request to look the sample and advise me what we should to do to bring close match with SAV output. 
Once again thanks for reply

**Comment by shawnhar** [comment]
> the output of Win2D is not close to SVG file. The direction of the gradient is different direction

Sure.  This is because Win2D and SVG are not the same thing.   Their capabilities are equivalent, but parameters don't always have the same meaning or order, so you can't just feed the same numbers into both and expect to get identical results.

You can either adjust parameters like the direction of your gradient to match the Win2D specifies these things, or if you want to draw SVG format data, I suggest taking a look at the SVG rendering feature that comes built in to Win2D.

**Comment by clandrew** [comment]
It looks like the problem is not the gradient, but that your geometry is incorrectly specified. If we forget even doing a radial gradiant at all and just fill the geometry with a solid black fill, we get:

![diamond](https://user-images.githubusercontent.com/8118775/30075349-a0e97c22-9229-11e7-869a-2007ad4ed4bb.png)

It looks like the geometry is drawn in four slices. If you want to draw it that way, you can fix it up with a 
```
pathBuilder.AddLine(new Vector2(end.X, start.Y));
```
or
```
pathBuilder.AddLine(new Vector2(start.X, end.Y));
```
before ending each figure. Just note that if you do it that way, since your geometry consists of four figures, there may be a seam in between the four slices depending on how you decide to draw it.
So, an even better way if you don't want a seam is to draw the whole geometry as one figure.

**Comment by AghaKhan** [comment]
Dear ClAndrew:

Thank you very much putting me in right direction.

I added pathBuilder.AddLine(new Vector2(end.X, start.Y));

But it didn't work but your suggestion put me in correct path.

Now I created one path (before there were 4 paths) where I have provide continuous AddCubicBezier function


Vector2 start = new Vector2(414f, 452.1f);
Vector2 ctrl0 = new Vector2(0f, 15.2f) + start;
Vector2 ctrl1 = new Vector2(-72.5f, 27.6f) + start;
Vector2 end = new Vector2(-162f, 27.6f) + start;


CanvasPathBuilder pathBuilder = new CanvasPathBuilder(sender);

pathBuilder.BeginFigure(start);
pathBuilder.AddCubicBezier(ctrl0, ctrl1, end);
change ctrl0, ctrl1, end values and add
pathBuilder.AddCubicBezier(ctrl0, ctrl1, end);
until our path is completed.
and last line should be

pathBuilder.EndFigure(CanvasFigureLoop.Closed); // last Z

CanvasGeometry geometry = CanvasGeometry.CreatePath(pathBuilder);
ds.FillGeometry(geometry, _canvasRadialGradientBrush);

it worked fine.
Thanks for help.
Best regards
Agha Khan








________________________________

**Comment by clandrew** [comment]
The actual placement of the extra line segment points is a bit finicky. All the same, I'm glad you fixed the geometry to work!

Since I don't see an additional action for us, I will close this bug but feel free to get in touch if you need anything further.
